### PR TITLE
[NFC][SYCL] Make clang-format tolerate more LIT checks

### PR DESCRIPTION
**Pull request by MrSidims**
Signed-off-by: Dmitry Sidorov <dmitry.sidorov@intel.com>

### CODE REVIEW

**Comment by bader** [comment]
@MrSidims, @romanovvlad, the description of the [PR](https://github.com/intel/llvm/pull/2149) added this exception promises something more than what we have implemented.

> But disable the limit for comments containing specific keywords:
> RUN, FAIL, REQUIRES, UNSUPPORTED, CHECK, expected-*

Do we want to fix the implementation to match the description?

**Comment by MrSidims** [comment]
> @MrSidims, @romanovvlad, the description of the [PR](https://github.com/intel/llvm/pull/2149) added this exception promises something more than what we have implemented.
> 
> > But disable the limit for comments containing specific keywords:
> > RUN, FAIL, REQUIRES, UNSUPPORTED, CHECK, expected-*
> 
> Do we want to fix the implementation to match the description?

I do believe, that the intention was to enable clang-format for C++ code in the tests. I'm not sure why we would like to follow clang-format for check/run strings and LLVM IR itself.

**Comment by bader** [comment]
> > @MrSidims, @romanovvlad, the description of the [PR](https://github.com/intel/llvm/pull/2149) added this exception promises something more than what we have implemented.
> > > But disable the limit for comments containing specific keywords:
> > > RUN, FAIL, REQUIRES, UNSUPPORTED, CHECK, expected-*
> > 
> > 
> > Do we want to fix the implementation to match the description?
> 
> I do believe, that the intention was to enable clang-format for C++ code in the tests. I'm not sure why we would like to follow clang-format for check/run strings and LLVM IR itself.

I think you are missing my point.
According to the description, the line limit should already be disabled for such comment.
But implementation doesn't match the description: if I read regular expression correctly, it disables the limit for the lines where specific keywords are followed by the space (e.g. word must end with it and not just contain) + there must be a colon after the keyword.

Another problem with this whole approach is that tests must use only standard FileCheck keywords and I don't see this rule to be harden by CI or some other way.

**Comment by bader** [comment]
> if I read regular expression correctly, it disables the limit for the lines where specific keywords are followed by the space (e.g. word must end with it and not just contain) + there must be a colon after the keyword.

"(RUN|FAIL|REQUIRES|UNSUPPORTED|CHECK) *:" 

I think this matches a keyword followed by a colon with 0 or more spaces between them. Right?
Anyway, this doesn't match the description of the [PR](https://github.com/intel/llvm/pull/2149) and we probably should fix this mismatch instead.

File: sycl/test/.clang-format [hunk 1]
```diff
@@ -1,2 +1,2 @@
 BasedOnStyle: LLVM
-CommentPragmas: "(RUN|FAIL|REQUIRES|UNSUPPORTED|CHECK) *:|expected-"
+CommentPragmas: "(RUN|FAIL|REQUIRES|UNSUPPORTED|CHECK|CHECK-NOT|CHECK-DAG|CHECK-SAME|CHECK-NEXT|CHECK-COUNT|CHECK-LABEL|) *:|expected-"
```
Comment (MrSidims) [inline sycl/test/.clang-format:2]:
```suggestion
CommentPragmas: "(RUN|FAIL|REQUIRES|UNSUPPORTED|CHECK|)*:|expected-"
```

**Comment by MrSidims** [comment]
@bader please check out the newest approach, hope I did get you correctly.
I've also noted, that sub_buffer.cpp failed in pre-commit, investigating.

File: sycl/test/.clang-format [hunk 2]
```diff
@@ -1,2 +1,2 @@
 BasedOnStyle: LLVM
-CommentPragmas: "(RUN|FAIL|REQUIRES|UNSUPPORTED|CHECK) *:|expected-"
+CommentPragmas: "(RUN|FAIL|REQUIRES|UNSUPPORTED|CHECK)*:|expected-"
```
Comment (bader) [inline sycl/test/.clang-format:2]: I don't think this is the right fix. 
My interpretation of this regex is "match 0 or more instances of any keywords followed by a colon" (ignoring "expected-" part).
I would expect the regex to look like:
```suggestion
CommentPragmas: "RUN|FAIL|REQUIRES|UNSUPPORTED|CHECK|expected-"
```

Comment (MrSidims) [reply sycl/test/.clang-format]: My interpretation was (is?) "match 0 or more instances of any keywords followed by a colon (anything can be put between the keyword and the colon) or match expected- (which is not followed by a colon)"

**Comment by MrSidims** [comment]
@bader clang-format there has passed: https://github.com/intel/llvm/runs/4651131820?check_suite_focus=true

**Comment by bader** [comment]
Good.

**Review by bader — APPROVED** [review]
Good enough for me. It's not perfect, but I don't think it's worth improving it before we see the need for this.
I'd like original author (@romanovvlad) to approve.

**Comment by MrSidims** [comment]
@romanovvlad could you please take a look?

**Comment by romanovvlad** [comment]
> Good enough for me. It's not perfect, but I don't think it's worth improving it before we see the need for this. I'd like original author (@romanovvlad) to approve.

Agree. I'm OK with current solution.

**Review by romanovvlad — APPROVED** [review]

**Pull request merged**